apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -338,21 +338,23 @@ def read(
         _util.check_unpartitioned(partitions)
         self._check_open_read()
 
+        handle = self._handle._handle
+
         ts = None
-        if self._handle._handle.timestamp is not None:
-            ts = (0, self._handle._handle.timestamp)
+        if handle.timestamp is not None:
+            ts = (0, handle.timestamp)
 
         sr = clib.SOMADataFrame.open(
-            uri=self._handle._handle.uri,
+            uri=handle.uri,
             mode=clib.OpenMode.read,
-            platform_config=platform_config or {},
+            platform_config=platform_config or handle.config(),
```

Comment:
shouldn't these be merged?  I'm thinking about the case where:
* handle.config() has the context-provided configuration (from open)
* user provides a platform_config to override something (e.g., buffer size)

I'm essentially suggesting that the clib `platform_config` argument should be:
>  platform_config=handle.config() | (platform_config or {})

Reply:
I think I understand. Do you mean `platform_config = handle.config().update(platform_config or {})`? The `handle.config()` (config that's passed to `open`) gets updated with any new key-value pairs in `platform_config` (config that's passed to `read`) but otherwise retains what's originally there.

Reply:
yep!  The read should use the union of the config, prioritizing the read() argument if specified.

@thetorpedodog - do you concur?

Reply:
> I think I understand. Do you mean `platform_config = handle.config().update(platform_config or {})`? The `handle.config()` (config that's passed to `open`) gets updated with any new key-value pairs in `platform_config` (config that's passed to `read`) but otherwise retains what's originally there.

> yep! The read should use the union of the config, prioritizing the read() argument if specified.

Well, we wouldn’t want to do `.update()` since that would end up permanently changing whatever gets returned from `handle.config()`, but yes, the union with passed-in value taking priority is what I would expect from a call like this.

Reply:
Thanks for the heads up. I have made a deep copy of the `handle.config()` before updating it. I tried to use `{**handle.config(), **(platform_config or {})}` but mypy was giving me a cryptic typing error.